src/fiber/channel.rs

```diff
@@ -1126,7 +1212,16 @@ where
             ))
             .expect(ASSUME_NETWORK_ACTOR_ALIVE);
         state.save_remote_nonce_for_raa();
-
+        if state
+            .tlc_state
+            .all_tlcs()
+            .filter(|tlc| tlc.is_offered())
+            .count()
+            > 0
+        {
+            state.tlc_state.set_waiting_ack(true);
```

Comment:
Why are we only expecting ack if there are offered TLCs? Say, A sent B a payment, and everything goes smoothly, B finally wants to remove this TLC with success, in this case, B is also expecting, a RevokeAndAck from A. It is hard for me to imagine any case where we send a `CommitmentSigned` and not expecting a `RevokeAndAck`.